### PR TITLE
docker dev env

### PR DESCRIPTION
**Pull request by smalho01**
This Pull Request touches multiple repositories and assumes a specific project structure as defined by the readme file in this pr under "End-To-End DRLS PAS Docker configuration". To review this, please also checkout the dockerDev branches for  [test-ehr](https://github.com/HL7-DaVinci/test-ehr/pull/59), [CRD](https://github.com/HL7-DaVinci/CRD/pull/252), [ crd-request-generator](https://github.com/HL7-DaVinci/crd-request-generator/pull/44), [dtr](https://github.com/HL7-DaVinci/dtr/pull/163), [prior-auth](https://github.com/HL7-DaVinci/prior-auth/pull/78),[ prior-auth-client](https://github.com/HL7-DaVinci/prior-auth-client/pull/2), [fhir-x12,](https://github.com/HL7-DaVinci/fhir-x12/pull/7) and [fhir-x12-frontend](https://github.com/HL7-DaVinci/fhir-x12-frontend/pull/2)

Published Docker Images for production set up are hosted here: https://hub.docker.com/repositories/docker/smalho01234. The hope is that these images will eventually make their way over to the logicahealth docker hub organization

### CODE REVIEW

File: src/main/java/org/hl7/davinci/priorauth/App.java [hunk 1]
```diff
@@ -108,6 +123,9 @@ public static ModelResolver getModelResolver() {
    * @return true if debugMode is true, false otherwise
    */
   public static boolean isDebugModeEnabled() {
-    return debugMode;
+    System.out.println("4-1debug mode in isDebugEnabled is: " + debugMode);
+    System.out.println("4-2debug mode in isDebugEnabled is: " + config.getProperty("debugMode"));
+    return true;
+    // return debugMode;
```
Comment (smalho01) [inline src/main/java/org/hl7/davinci/priorauth/App.java:129]: hard coding debug mode to always true as temporary work around - debugMode variable changes to false at end of Gradle bootRun with springboot version upgrade. Not sure why this behavior occurs

**Comment by smalho01** [comment]
I can't get get dtr to authenticate with pas locally or as part of this docker set up. There's no issues with the logica servers or with the PAS AEGIS tests so this is not a bug but user error on my part, I must be messing up the registration step or something. If anyone reviewing this is able to test authentication between pas and dtr and get it to work, please respond to this comment

File: docker-sync.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+version: "2"
+options:
+  compose-file-path: './docker-compose-dev.yml'
+syncs: 
+  pas_dev_test-ehr-sync:
+    src: ../test-ehr
+    sync_excludes: ['.gradle', 'build', 'target', 'logs']
```
Comment (smalho01) [inline docker-sync.yml:7]: If I am missing any other files/directories that should not be synced between the local machine and container, let me know. Those files need to be specified here and in the docker-compose.yml volumes (not specifying in volume will result in the sync strategy removing the file/folder from the container all together) for all repos

Examples include: build files, binaries, dependencies, database files, etc.

File: DockerDevSetupGuideForMacOS.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+5. Click anywhere in the row for William Oster.
+6. Click **Submit** at the bottom of the page.
+7. After several seconds you should receive a response in the form of two **CDS cards**:
+    - **Respiratory Assist Device**
+    - **Positive Airway Pressure Device**
+8. Select **Order Form** on one of those CDS cards.
+9. If you are asked for login credentials, use **alice** for username and **alice** for password.
+10. A webpage should open in a new tab, and after a few seconds, a questionnaire should appear.
+11. Fill out questionnaire and hit next
+12. Submit PAS Request to http://localhost:9015/fhir
```
Comment (smalho01) [inline DockerDevSetupGuideForMacOS.md:335]: I can't get get dtr to authenticate with pas locally or as part of this docker set up or the production docker set ups. There's no issues with the logica servers or with the PAS AEGIS tests so I don't think this is a bug but rather something that I must missing. The issue is not a networking issue either as dtr can hit the pas metadata endpoint with no issues. If anyone reviewing can get dtr and pas to authenticate, let me know by responding to this comment.

File: Dockerfile.tmpl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+RUN apt-get source git && \
+    cd git-2.*.*/ && \
+    sed -i -- 's/libcurl4-gnutls-dev/libcurl4-openssl-dev/' ./debian/control && \
+    sed -i -- '/TEST\s*=\s*test/d' ./debian/rules && \
+    dpkg-buildpackage -rfakeroot -b -uc -us && \
+    dpkg -i ../git_*ubuntu*.deb
+
+COPY ./.cnab/app/porter.yaml ${BUNDLE_DIR}/porter.yaml
+COPY ./docker-compose-porter.yml ${BUNDLE_DIR}
+COPY ./.env ${BUNDLE_DIR}
```
Comment (smalho01) [inline Dockerfile.tmpl:42]: Potential Security Concern: My VSAC API Key is added to the Invocation Image that is published and hence can be viewed by someone running the Porter production set up who execs into the Invocation Image Container and views the contents of the .env file

File: DockerDevSetupGuideForMacOS.md (hunk 3 above)
Comment (smalho01) [reply DockerDevSetupGuideForMacOS.md:336]: http vs https? Only difference I can think of between running locally and running on logicahealth

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+  prior-auth-client: # Name of our service
+    image: smalho01234/prior-auth-client
+    container_name: pas_prod_prior-auth-client
+    ports: # Port binding to host from docker container
+      - "9090:9090" # Bind port 3000 of host to 3000 of container
+
+
+  # Create fhir-x12 converter container
+  fhir-x12: # Name of our service
```
[the remark this answers is not in the document]
Comment (garypgryan) [reply docker-compose.yml:79]: Is there a way to compile FHIR<=> X12 to binary?   I saw some called Deno which might work if it's not a huge lift.  https://www.infoq.com/news/2021/02/deno-compiles-native-binaries/

**Review by garypgryan — APPROVED** [review]
Looks great.   I don't know if you can compile the typescript to binary or run it through an code obsfucator.  We should discuss.

**Pull request merged**